src/components/MessagesList/MessagesList.vue

```diff
@@ -583,6 +660,121 @@ export default {
 				this.displayMessagesLoader = false
 				this.previousScrollTopValue = scrollTop
 			}
+
+			this.debounceUpdateReadMarkerAfterScroll()
+		},
+
+		/**
+		 * Find the next message element following the given message DOM element.
+		 *
+		 * This traverses the next messages and message groups to find the next one.
+		 *
+		 * @param {object} messageEl DOM element for message to start with
+		 * @returns {object} DOM element for the next message or null if none found
+		 */
+		findNextMessageElement(messageEl) {
+			// pick the previous message
+			let searchEl = messageEl.nextElementSibling
```

Comment:
Does this also work with the "group by author" feature we have and with system messages in between?

Reply:
it works with the message groups, if you read further below you can see how it's exiting the current group and entering the next one.

for system messages I'll double check

Reply:
I did a test with a lot of system messages, then scrolled down and waited 3s to have the marker update, then scrolled up again:
<img width="312" alt="image" src="https://user-images.githubusercontent.com/277525/109505653-1a9ec900-7a9d-11eb-9c93-733e42b0656a.png">

this means the detection within those groups also work.
they actually use the same classes "message-group" and "message", so this was expected